conan/all/test_package/main.c

```diff
@@ -0,0 +1,45 @@
+#include <sqlite3mc/sqlite3mc.h>
+#include <stdio.h>
+
+int main(int argc, char *argv[])
+{
+    sqlite3 *db;
+    const char *key = "password";
+
+    int rc = sqlite3_open("test.db", &db);
+    if (rc != SQLITE_OK)
+    {
+        fprintf(stderr, "Failed to open database: %s\n", sqlite3_errmsg(db));
+        sqlite3_close(db);
+        return 1;
+    }
+
+    rc = sqlite3_key(db, key, strlen(key));
```

Comment:
`sqlite3_key` can only fail, if it is called for an in-memory or temporary database, or if the URI specifies an invalid cipher scheme. It does **not** fail if an invalid password was given. A wrong password will be detected only, after the database file was actually accessed, that is, after any SQL DML command was issued.

For a newly created database file it will almost never fail.

Reply:
I have tried the following:
1. Creating a db
2. Use sqlite3_key
3. Creating a users table and insert one user
4. Close the db
5. Open the db again
6. Use sqlite3_key with a wrong password

But it's returning with SQLITE_OK. Am I missing something?
If I try to access an already existing encrypted db with another exe, its working, I can't access it without the correct password. But I can't reproduce the same within one test program.

Reply:
> You mean sqlite3_key returns an error??? That can't be. An error is returned when you try to read or write data.

You're right, the query returned the error, not the sqlite3_key.

> Just issue a query after calling sqlite3_key. This should result in an error if the wrong password was provided.

Thanks, it's working now. I'll update the PR this evening.